packages/infra/lib/api-stack/cw-directory-rebuilder.ts

```diff
@@ -0,0 +1,69 @@
+import { Duration } from "aws-cdk-lib";
+import { SnsAction } from "aws-cdk-lib/aws-cloudwatch-actions";
+import { IVpc } from "aws-cdk-lib/aws-ec2";
+import { Function as Lambda } from "aws-cdk-lib/aws-lambda";
+import { Construct } from "constructs";
+import { EnvConfig } from "../../config/env-config";
+import { getConfig } from "../shared/config";
+import { LambdaLayers } from "../shared/lambda-layers";
+import { createScheduledLambda } from "../shared/lambda-scheduled";
+
+export type CwDirectoryRebuilderProps = {
+  stack: Construct;
+  lambdaLayers: LambdaLayers;
+  vpc: IVpc;
+  apiAddress: string;
+  alarmSnsAction?: SnsAction;
+};
+
+function getSettings(
+  props: CwDirectoryRebuilderProps,
+  config: NonNullable<EnvConfig["cwDirectoryRebuilder"]>
+) {
+  return {
+    ...props,
+    name: "ScheduledCwDirectoryRebuilder",
+    lambdaMemory: 256,
+    lambdaTimeout: Duration.seconds(900), // How long can the lambda run for, max is 900 seconds (15 minutes)
```

Comment:
Pretty sure the base scheduled lambda impl just calls the endpoint and doesn't wait for it to return, so this can be just a few seconds (maybe 30?)